### DISCUSSION TITLE
Event on checkbox input trigger twice :/

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2189] [Q&A]
Hi everyone,

I'm using `bind()` quite a lot with Bryhton and so far no problem until now.

I've got
```python

def foobar(ev):
    print('test')

document['aCheckBox'].bind('input', foobar)
```
So when I tick this checkbox in the browser it trigger twice and I get twice in the console 'test'

I've searched the WWW and it seem I'm not the only one... I've tried all the solution but it still happening... any ideas ?

I've tried also with the following event : click, change, input. but still the problem remain,

**Comment by BobSquarePants** [comment]
OK, I've a way

```python
def foobar(ev):
    ev.stopImmediatePropagation()
    print('test')
```

Thanks to: https://javascript.info/bubbling-and-capturing

This is insane, on all the others elements no problem at all when binding an a event but just on the checkbox this cause a double trigger.. 🤦‍♂️ 

So side question, where that weird behavior come from ? Because I would like to inform the dev team about it.

DOM Standard ?

Cheers.

**Comment by PierreQuentel** [comment]
This is strange. Can you post an example where you get this behavior ?

In this example the callback function is called only once

```xml
<script type="text/python">
from browser import bind

@bind('#test', 'input')
def test(ev):
  print('ok')
</script>

<input type="checkbox" id="test">
```

**Reply by BobSquarePants** [reply]
Hi @PierreQuentel , sure I will post a complete example asap.
meanwhile you might try with the <input> element with a <div> not sure it will reproduce what I get.
```html
<div><input type="checkbox" id="test"></div>
```

**Reply by BobSquarePants** [reply]
@PierreQuentel , I've done some test on blank page and indeed it work how it should be.
I don't have the time to debug right now my code, but I assume that I've done a double binding... (by reloading one of my Brython script).

If I found my bug, I will reopen this topic. Thanks again :)
